services/QuillLMS/app/services/google_integration/users/classroom_data_adapter.rb

```diff
@@ -0,0 +1,36 @@
+module GoogleIntegration
+  module Users
+    class ClassroomDataAdapter
+      attr_accessor :user, :data
+
+      def initialize(user, data)
+        @user = user
+        @data = data
+      end
+
+      def run
```

Comment:
You might consider removing the mutation here, i.e. the sole use of `run` here is to mutate the `data` attribute that you are passing in the initializer. So `data` will be very different depending on whether this has been `run` or not, which may not be obvious or known to a person using this object. 

You could have a memoizable attribute called `cleaned_data` or `parsed_data` (or something, even `run`). This would remove the mutation.

```ruby
def parsed_data
  @parsed_data ||= do
     ...generate the correct data hash
   end
end

Reply:
@dandrabik Does this work

Reply:
I was debating on using `tap` but opted for a clean hash so we are more transparent about the adapter.

Reply:
On second thought.  I think I see what you mean.